From a7196f60905d9b078d7cc806629849402cef200a Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Mon, 8 Jan 2024 12:44:41 +0100
Subject: [PATCH 0835/1141] media: stm32: dcmipp: isp: add a new pad for
 parameters injection

Add a new SINK pad to be linked to the ISP parameter video device
for control of the ISP parameters.

Change-Id: Ic23d5fa7c2b8d007da1823db7c5cc614f423022e
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/353230
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-isp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-isp.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-isp.c
@@ -1135,10 +1135,11 @@ struct dcmipp_ent_device *dcmipp_isp_ent
 	ret = dcmipp_ent_sd_register(&isp->ved, &isp->sd,
 				     v4l2_dev,
 				     entity_name,
-				     MEDIA_ENT_F_PROC_VIDEO_PIXEL_FORMATTER, 3,
-				     (const unsigned long[3]) {
+				     MEDIA_ENT_F_PROC_VIDEO_PIXEL_FORMATTER, 4,
+				     (const unsigned long[4]) {
 				     MEDIA_PAD_FL_SINK,
 				     MEDIA_PAD_FL_SOURCE,
+				     MEDIA_PAD_FL_SINK,
 				     MEDIA_PAD_FL_SOURCE,
 				     },
 				     &dcmipp_isp_int_ops, &dcmipp_isp_ops,
